### PR TITLE
Site resdesign/rebuild - first pass

### PR DESCRIPTION
**Pull request by jperezlatimes**
Initial design pass.  Header, navigation buttons, and main blogroll

some reappropriation/refactoring of old code

I'm mostly interested on any feedback you have on the hsinsider_get_school function in template_tags.php.  That file is full of old code renamed to current standards.  Most of it is alright, but I had to wind up refactoring that function.

### CODE REVIEW

File: client/grunt/concat.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
   "dist": {
     "src": [
     	"client/js/lib/*.js",
+    	"client/js/lib/*.*.min.js",
```
Comment (Charlex) [inline client/grunt/concat.json]: Does this mean you're minifying `.min.js` files?

File: client/js/navigation.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+		menuContainer.show( "slide", { direction: "right" }, 500 );
+		
+		$( 'html' ).click( 'menuOffClick' );
+	} );
+	
+	$( 'html' ).click( function( e ) {
+
+		if( thisClick ) {
+			thisClick = false;
```
Comment (Charlex) [inline client/js/navigation.js]: What are you using `thisClick` and `menuContainer ` for? Global vars always make me ask

File: client/sass/_menu.scss [hunk 3]
```diff
@@ -1,91 +1,128 @@
...
-		border-radius: 0px;
-
-		.navbar-brand {
-			display: none;
-			padding: 0px;
-		}
-
-		.navbar-toggle {
-			@include hsinsider-border-radius( 0px );
+menuwrapper {
```
Comment (Charlex) [inline client/sass/_menu.scss]: Should this be a class?

File: client/sass/_structure.scss [hunk 4]
```diff
@@ -19,7 +19,6 @@
 
 html {
 	body{
-
 		header.site-header {
```
Comment (Charlex) [inline client/sass/_structure.scss]: Can this die?

File: client/sass/_variables.scss [hunk 5]
```diff
@@ -7,13 +7,16 @@ $bootstrap-sass-asset-helper: (twbs-font-path("") != unquote('twbs-font-path("")
...
+
+$offcanvas-animation-time: 0.25s;
 
 //== Colors
 //
 //## Gray and brand colors for use across Bootstrap.
 
 $gray-darker:            lighten(#000, 13.5%) !default; // #222
-$gray-dark:              lighten(#000, 20%) !default;   // #333
+$gray-dark:				 #202020;
```
Comment (Charlex) [inline client/sass/_variables.scss]: Lots o' white space here

File: client/sass/_variables.scss [hunk 6]
```diff
@@ -34,20 +40,19 @@ $highlight-color: #dcfba8;
...
+$link-hover-color:      darken( $highlight-color, 12% );
 
 
 //== Typography
 //
 //## Font, line-height, and color for body text, headings, and more.
-
-$font-family-sans-serif:  "Helvetica Neue", Helvetica, Arial, sans-serif !default;
-$font-family-serif:       Georgia, "Times New Roman", Times, serif !default;
+$font-family-sans-serif:  'Source Sans Pro', sans-serif;
```
Comment (Charlex) [inline client/sass/_variables.scss]: Can we throw in Helvetica, Arial as fallbacks?

File: client/grunt/concat.json (hunk 1 above)
Comment (jperezlatimes) [inline client/grunt/concat.json]: It's stitching together all js files and minifying the result.  So yes.

File: client/js/navigation.js (hunk 2 above)
Comment (jperezlatimes) [inline client/js/navigation.js]: This isn't my code and I have no *&%ing idea what thisClick is for.  menuContainer is some weird data object for the currently opened menu

File: client/sass/_menu.scss (hunk 3 above)
Comment (jperezlatimes) [inline client/sass/_menu.scss]: yes it should

File: client/sass/_structure.scss (hunk 4 above)
Comment (jperezlatimes) [inline client/sass/_structure.scss]: can what die?

File: header.php [hunk 7]
```diff
@@ -24,8 +24,3 @@
 		</div>
 		<?php endif; ?>
 		<div id="content" class="site-content container">
-			<div class="row hidden-sm hidden-xs">
-				<div class="col-sm-12">
-					<div class="leaderboard ad" id="div-gpt-ad-783778988016615787-1"></div>
-				</div>
-			</div>
```
Comment (Charlex) [inline header.php]: Take it you're mapping out the ads differently from how this is configged?

File: inc/assets.php [hunk 8]
```diff
@@ -24,6 +28,7 @@ function hsinsider_enqueue_assets() {
 	wp_register_script( 'google-maps', 'http://maps.googleapis.com/maps/api/js?key=' . hsinsider_site_config( 'api.googlemaps' ) . '&callback=initMap', array( 'hsinsider-global-js' ), '1.0', true );
 	
 	wp_enqueue_script( 'jquery' );
+	wp_enqueue_script( 'jquery-effects-slide' );
```
Comment (Charlex) [inline inc/assets.php]: Isn't slide already a feature of jquery?

File: inc/rewrites.php [hunk 9]
```diff
@@ -2,3 +2,31 @@
...
+
+	$posted_on = '<a href="' . esc_url( get_permalink() ) . '" rel="boomark">' . _( $time_string ) . '</a>';
+
+	$byline = '<a href="' . esc_url( get_author_posts_url( get_the_author_meta( 'ID' ) ) ) . '">' . esc_html( get_the_author() ) . '</a>';
+
+	echo '<div class="byline"><span class="posted-on">' . $posted_on . '</span>&nbsp;|&nbsp;<span class="post-author">' . $byline . '</span></div>';
+}
+
+function hsinsider_excerpt_more( $more ) {
+	return '...';
```
Comment (Charlex) [inline inc/rewrites.php]: Can the html entity `&ellipsis;` be used instead? NBD, just fo style

File: client/sass/_variables.scss (hunk 5 above)
Comment (jperezlatimes) [inline client/sass/_variables.scss]: this is bootstraps doing.  it's default variables file is setup that way

File: header.php (hunk 7 above)
Comment (jperezlatimes) [inline header.php]: yup

File: inc/template-tags.php [hunk 10]
```diff
@@ -115,46 +115,43 @@ function hsinsider_category_transient_flusher() {
...
+		$terms = get_the_terms( $post->ID, 'school' );
+
+		if( !empty( $terms ) || is_array( $terms ) ) {
+			$term_id = $terms[0]->term_id;
+			$term = ( is_object( wpcom_vip_get_term_by( 'id', $term_id, 'school' ) ) ) ? wpcom_vip_get_term_by( 'id', $term_id, 'school' ) : false;
+		}
 	}
-		
 	return $term;
 }
```
Comment (Charlex) [inline inc/template-tags.php]: So many if statements. A little hard to read. 

Maybe this is the way this must be, but if you're in a refactoring mood, I'd flag this as a candidate.

Comment (jperezlatimes) [inline inc/template-tags.php]: This is refactored.  at least the hsinsider_get_school function is.  I agree, on the plethora of if statements though.

**Comment by Charlex** [comment]
Why is screenshot.png called screenshot.png? Can we call this hsinsider_logo.png or something else?

**Comment by jperezlatimes** [comment]
It has to be called screenshot or wordpress won't recognize it

File: static/fonts/bootstrap/glyphicons-halflings-regular.svg [hunk 11]
```diff
@@ -0,0 +1,229 @@
+<?xml version="1.0" standalone="no"?>
```
Comment (Charlex) [inline static/fonts/bootstrap/glyphicons-halflings-regular.svg]: Are we using both glyphicons and font-awesome? Seems like we should pick one and go with that, even if it means taking away a part of bootstrap's native icon stuff.

File: static/js/global.js [hunk 12]
```diff
@@ -2213,6 +2213,179 @@ function codeAddress() {
 }
 
 /**
+ * Controls the Navigation Menu and Search Form
```
Comment (Charlex) [inline static/js/global.js]: Is this deja vu all over again? Isn't all of this in `client/js/navigation.js`?

File: static/fonts/bootstrap/glyphicons-halflings-regular.svg (hunk 11 above)
Comment (jperezlatimes) [inline static/fonts/bootstrap/glyphicons-halflings-regular.svg]: Actually we have glyphicons, fontawesome, and the datadesk cookbook icon sets

File: client/js/navigation.js (hunk 2 above)
Comment (Charlex) [inline client/js/navigation.js]: Gross.

File: client/sass/_structure.scss (hunk 4 above)
Comment (Charlex) [inline client/sass/_structure.scss]: There's nothing in `header.site-header {}`

File: client/sass/_variables.scss (hunk 5 above)
Comment (Charlex) [inline client/sass/_variables.scss]: k

File: static/fonts/bootstrap/glyphicons-halflings-regular.svg (hunk 11 above)
Comment (Charlex) [inline static/fonts/bootstrap/glyphicons-halflings-regular.svg]: Seems like overkill :confused:

**Comment by jperezlatimes** [comment]
:godmode:

**Pull request merged**